app/frontend/src/features/auth/password/ChangePasswordPage.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  makeStyles,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from "@material-ui/core";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+
+import Alert from "../../../components/Alert";
+import Button from "../../../components/Button";
+import PageTitle from "../../../components/PageTitle";
+import TextField from "../../../components/TextField";
+import { service } from "../../../service";
+
+interface ChangePasswordVariables {
+  oldPassword?: string;
+  newPassword?: string;
+}
+
+interface ChangePasswordFormData extends ChangePasswordVariables {
+  passwordConfirmation?: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+  form: {
+    "& > * + *": {
+      marginBlockStart: theme.spacing(1),
+    },
+  },
+  infoText: {
+    marginBlockEnd: theme.spacing(1),
+  },
+}));
+
+export default function ChangePasswordPage() {
+  const classes = useStyles();
+  const theme = useTheme();
+  const isMdOrWider = useMediaQuery(theme.breakpoints.up("md"));
+
+  const {
+    errors,
+    getValues,
+    handleSubmit,
+    register,
+  } = useForm<ChangePasswordFormData>({
+    mode: "onBlur",
+  });
+
+  const { error, isLoading, isSuccess, mutate: changePassword } = useMutation<
+    Empty,
+    GrpcError,
+    ChangePasswordVariables
+  >(({ oldPassword, newPassword }) =>
+    service.account.changePassword(oldPassword, newPassword)
+  );
+
+  const onSubmit = handleSubmit(({ oldPassword, newPassword }) => {
+    changePassword({ oldPassword, newPassword });
+  });
+
+  return (
+    <>
+      <PageTitle>Change password</PageTitle>
+      {error && <Alert severity="error">{error.message}</Alert>}
+      {isSuccess && (
+        <Alert severity="success">
+          Your password change has been processed. Check your email for
+          confirmation.
+        </Alert>
+      )}
+      <Typography className={classes.infoText} variant="body1">
```

Comment:
I noticed we start using `Typography` more for other heading levels. I guess since all the styles seem to be standardised/controlled through the theme's typography variants, `BodyText` feels like a redundant wrapper to me at this point  🤔